Add automatic secure boot docs (#282)
metal3-io/metal3-docs: docs/user-guide/src/bmo/automatic_secure_boot.md

# Automatic secure boot

> The automatic secure boot feature allows enabling and disabling UEFI (Unified Extensible Firmware Interface) secure boot when provisioning a host. This feature requires supported hardware and compatible OS image. The current hardwares that support enabling UEFI secure boot are `iLO driver`, `iRMC driver` and `Redfish driver`.

Reviewer: nit: not sure why use ticks around drivers, maybe you meant `iLO`, `iRMC` and `Redfish` drivers?

> Check also

Reviewer 2: ```suggestion
Check also:
```

> - [Ironic UEFI secure-boot](https://docs.openstack.org/ironic/latest/admin/security.html#uefi-secure-boot-mode)

Reviewer: "secure boot" (without a dash, it's not a compound word). same below

Reviewer 2: Done previously

> - [Wikipeda UEFI secure-boot](https://en.wikipedia.org/wiki/UEFI#SECURE-BOOT)

Reviewer 2: ```suggestion
- [Wikipedia UEFI secure-boot](https://en.wikipedia.org/wiki/UEFI#SECURE-BOOT)
```

> ## Advantages/Benefits

Reviewer 2: Since it is answering "why" , we could rename it to:
```suggestion
## Why do we need a secure boot
```

Author: that was based on this review https://github.com/metal3-io/metal3-docs/pull/282#discussion_r982021125

Reviewer 2: hmm, I am not sure really if advantages/benefits fits better than this..

Author: @kashifest what do you think ? not sure if we have a common pattern for other docs

Reviewer 3: I am ok with this change

Reviewer 2: Considering this has been reviewed by some folks, I took care of it addressing this and the below comment since @mboukhalfa is away and we are tight on populating as much content as possible for the user guide. Any left out comments/concerns can be addressed in a follow-up PR.

> Why we need it?

Reviewer 3: ```suggestion
## Advantages/Benefits?
```

We need the automatic secure boot when provisioning a host with high security requirements. Based on checksum and signature, the secure boot protects the host from loading malicious code in the boot process before loading the provisioned operating system.

> ## Usage

Reviewer 2: and maybe:
```suggestion
## How to use it
```

Author: same https://github.com/metal3-io/metal3-docs/pull/282#discussion_r982022576

> How to use it?

Reviewer 3: ```suggestion
## Usage
```

> To enable Automatic secure boot check that the hardware is supported then specify the value `UEFISecureBoot` for `bootmode` in the bmh custom resource. Check the example:

Reviewer: typo: bootMode

Reviewer: I'd also spell out BareMetalHost since the "bmh" shortening is not used in this document

Author: thanks done

> To enable Automatic secure boot check that the hardware is supported then specify the value `UEFISecureBoot` for `bootmode` in the BareMetalHost custom resource. Check the example:

Reviewer 2: Thank you, done

```YAML
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: node-1
spec:
  online: true
  bootMACAddress: 00:5c:52:31:3a:9c
  bootMode: UEFISecureBoot
  ...
```

> This will enable UEFI in before booting the instance and disabled when deprovisioned. Note that the default value for `bootMode` is `UEFI`.

Reviewer 2: ```suggestion
This will enable UEFI before booting the instance and disable it when deprovisioned. Note that the default value for `bootMode` is `UEFI`.
```

# Automatic secure boot

The automatic secure boot feature allows enabling and disabling UEFI secure boot on and off when provisioning a host.

> why we need it?

Reviewer: This is probably going to be formatted as a wall of text, not the way you expect

> We need automatic secure boot when provisoing a host with high security requirements 

Author: thanks I will expand this and add ref from wiki

> We need automatic secure boot when provisoing a host with high security requirements 

Author: added ref from ironic docs and wiki

how to use it?

> To enable Automatic secure boot we need to specify the value `UEFISecureBoot` for `bootmode` in the bmh custom resource 

Reviewer: `bootMode`

Somewhere here we need to explain that:

1. It only works with some drivers (I think only redfish-based, definitely not IPMI).
2. Secure boot is enabled before booting into the deployed instance and disabled when the ramdisk is running and on tear down.

Author: Is the secure boot disabled on deprovisioning or once the instance booted ? should I specify ramdisk image here ?

Reviewer 2: I have added the second point as a note to the user as is.
